### ISSUE TITLE
Is it possible to get the title of the videos of a playlist without downloading them?

### ISSUE THREAD

**Issue by elieobeid7** [pytube/pytube#349]

**Comment by gregdan3** [comment]
Do you mean, to get the titles of each video, but not download the videos themselves?

You should be able to, once you have a URL to the video.
The Playlist object has a method to gather those URLs, and then you can create YouTube objects out of each. No need to download each video. 

```
from pytube import YouTube, Playlist
pl = Playlist("your playlist here")
pl.populate_video_urls()  # fills the pl.video_urls with all links from the playlist
urls = pl.video_urls

for url in urls:
    vid_title = YouTube(url).title  # here's what you want
```

**Comment by elieobeid7** [comment]
@gregdan3, Thank you for your reply, I really appreciate it.  Just tried that, I get

```

> Reloading due to file change: api.py
Serving on :8000...
Traceback (most recent call last):
  File "/home/elie/anaconda3/lib/python3.6/wsgiref/handlers.py", line 137, in run
    self.result = application(self.environ, self.start_response)
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/falcon/api.py", line 244, in __call__
    responder(req, resp, **params)
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/hug/interface.py", line 793, in __call__
    raise exception
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/hug/interface.py", line 766, in __call__
    self.render_content(self.call_function(input_parameters), context, request, response, **kwargs)
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/hug/interface.py", line 703, in call_function
    return self.interface(**parameters)
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/hug/interface.py", line 100, in __call__
    return __hug_internal_self._function(*args, **kwargs)
  File "api.py", line 38, in youtube_url
    pytube_test(url)
  

## heres the part related to pytube

File "api.py", line 16, in pytube_test
    pl.populate_video_urls()  # fills the pl.video_urls with all links from the playlist
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/pytube/contrib/playlist.py", line 95, in populate_video_urls
    link_list = self.parse_links()
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/pytube/contrib/playlist.py", line 61, in parse_links
    url = self.construct_playlist_url()
  File "/home/elie/Desktop/hug/venv/lib/python3.6/site-packages/pytube/contrib/playlist.py", line 36, in construct_playlist_url
    playlist_code = self.playlist_url.split('&list=')[1]
IndexError: list index out of range
127.0.0.1 - - [26/Dec/2018 22:34:37] "GET /api?url=https://www.youtube.com/watch?v=1cQh1ccqu8M&list=PLyjgHc47unfT3BIZo5uEt2a-2TWKy54sU HTTP/1.1" 500 59
```


Here's my code

```
def pytube_test(url):
    pl = Playlist(url)
    pl.populate_video_urls()  # fills the pl.video_urls with all links from the playlist
    urls = pl.video_urls
    for url in urls:
        vid_title = YouTube(url).title  # here's what you want
        print (vid_title)

@hug.get('/api')
def youtube_url(url):
    pytube_test(url)
```

And here's the url I'm testing with

https://www.youtube.com/watch?v=1cQh1ccqu8M&list=PLyjgHc47unfT3BIZo5uEt2a-2TWKy54sU

**Comment by gregdan3** [comment]
When I ran your code with that URL, I first got this error:
```
$ python c:/.../youtube/playlisttest.pyTraceback (most recent call last):
  File "c:/.../youtube/playlisttest.py", line 23, in <module>
    youtube_url('https://www.youtube.com/watch?v=1cQh1ccqu8M&list=PLyjgHc47unfT3BIZo5uEt2a-2TWKy54sU')
  File "c:/.../youtube/playlisttest.py", line 21, in youtube_url
    pytube_test(url)
  File "c:/CFRL/.../playlisttest.py", line 17, in pytube_test
    vid_title = YouTube(url).title  # here's what you want
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 88, in __init__
    self.prefetch_init()
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 97, in prefetch_init
    self.init()
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 133, in init
    mixins.apply_signature(self.player_config_args, fmt, self.js)
  File "C:\...\Python\Python37\lib\site-packages\pytube\mixins.py", line 49, in apply_signature
    signature = cipher.get_signature(js, stream['s'])
  File "C:\...\Python\Python37\lib\site-packages\pytube\cipher.py", line 250, in get_signature
    tplan = get_transform_plan(js)
  File "C:\...\Python\Python37\lib\site-packages\pytube\cipher.py", line 69, in get_transform_plan
    name = re.escape(get_initial_function_name(js))
  File "C:\...\Python\Python37\lib\site-packages\pytube\cipher.py", line 45, in get_initial_function_name
    return regex_search(pattern, js, group=1)
  File "C:\...\Python\Python37\lib\site-packages\pytube\helpers.py", line 41, in regex_search
    .format(pattern=p),
pytube.exceptions.RegexMatchError: regex pattern (yt\.akamaized\.net/\)\s*\|\|\s*.*?\s*c\s*&&\s*d\.set\([^,]+\s*,\s*(?P<sig>[a-zA-Z0-9$]+)\() had zero matches
```
This matches another issue, #312, which was already fixed #313.

So I updated pytube (I was running the latest version in pip, 9.3.5, but above is 9.3.6) with the following command:
`pip install git+https://github.com/nficano/pytube.git`
And got a new error, though this was expected:

```
$ python c:/.../youtube/playlisttest.py
Nickelback - How You Remind Me



Traceback (most recent call last):
  File "c:/.../youtube/playlisttest.py", line 23, in <module>
    youtube_url('https://www.youtube.com/watch?v=1cQh1ccqu8M&list=PLyjgHc47unfT3BIZo5uEt2a-2TWKy54sU')
  File "c:/.../youtube/playlisttest.py", line 21, in youtube_url
    pytube_test(url)
  File "c:/.../youtube/playlisttest.py", line 17, in pytube_test
    vid_title = YouTube(url).title  # here's what you want
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 88, in __init__
    self.prefetch_init()
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 96, in prefetch_init
    self.prefetch()
  File "C:\...\Python\Python37\lib\site-packages\pytube\__main__.py", line 162, in prefetch
    raise VideoUnavailable('This video is unavailable.')
pytube.exceptions.VideoUnavailable: This video is unavailable.
```

The second video in the playlist is blocked in my country. Ouch.
However, this DID work for the first video; catching the exception and continuing did not cause any more errors.
But, I can't seem to reproduce the error you got, either with my code or yours, though I haven't tested on python 3.6 which you seem to be using.
From what I can tell, the line that caused the error (.split('&list=')) should not fail in 3.6 or 3.7, as long as there is a string '&list=' in the url you pass. I'm not sure what happened, exactly.

Do you have any other information about what happened? And, have you tried running it without the api lines from hug?




EDIT: Are you sure you're passing the URL you gave me? The only way .split(delimiter) returns a single item list is if the delimiter is not in the string you pass it.
i.e.
```
a = "a very good string i should hope"
a.split('no delimiter here')
...
["a very good string i should hope"]
```

So doing a.split(...)[1] would give you an index out of range.

**Comment by elieobeid7** [comment]
@gregdan3 After upgrading everything worked, thank you so much.
